packages/prepo-shared-contracts/test/WithdrawTokens.test.ts

```diff
@@ -0,0 +1,129 @@
+      expect(await withdrawTokens.getNominee()).to.eq(ZERO_ADDRESS)
+    })
+  })
+
+  describe('# batchWithdrawERC20', async () => {
+    beforeEach(async () => {
+      await setupWithdrawTokens()
+      const mockERC20Factory = await smock.mock('ERC20Mintable')
+      firstMockERC20 = await mockERC20Factory.deploy('firstMockERC20', 'MERC20F')
+      await firstMockERC20.connect(owner).mint(withdrawTokens.address, parseEther('1'))
+      secondMockERC20 = await mockERC20Factory.deploy('secondMockERC20', 'MERC20S')
+      await secondMockERC20.connect(owner).mint(withdrawTokens.address, parseEther('1'))
+    })
+
+    it('reverts if not owner', async () => {
+      expect(await withdrawTokens.owner()).to.not.eq(user1.address)
+
+      await expect(
+        withdrawTokens.connect(user1).batchWithdrawERC20([firstMockERC20.address], [1])
+      ).revertedWith('Ownable: caller is not the owner')
+    })
+
+    it('reverts if array length mismatch', async () => {
+      expect([firstMockERC20].length).to.not.eq([1, 2].length)
+
+      await expect(
+        withdrawTokens.connect(owner).batchWithdrawERC20([firstMockERC20.address], [1, 2])
+      ).revertedWith('Array length mismatch')
+    })
+
+    it('reverts if amount > contract balance', async () => {
+      const contractFirstERC20BalanceBefore = await firstMockERC20.balanceOf(withdrawTokens.address)
+
+      await expect(
+        withdrawTokens
+          .connect(owner)
+          .batchWithdrawERC20([firstMockERC20.address], [contractFirstERC20BalanceBefore.add(1)])
+      ).revertedWith('ERC20: transfer amount exceeds balance')
+    })
+
+    it('withdraws if amount = contract balance', async () => {
+      const contractFirstERC20BalanceBefore = await firstMockERC20.balanceOf(withdrawTokens.address)
+      const ownerFirstERC20BalanceBefore = await firstMockERC20.balanceOf(owner.address)
+
+      await withdrawTokens
+        .connect(owner)
+        .batchWithdrawERC20([firstMockERC20.address], [contractFirstERC20BalanceBefore])
+
+      expect(await firstMockERC20.balanceOf(withdrawTokens.address)).to.eq(0)
+      expect(await firstMockERC20.balanceOf(owner.address)).to.eq(
+        ownerFirstERC20BalanceBefore.add(contractFirstERC20BalanceBefore)
+      )
+    })
+
+    it('withdraws if amount < contract balance', async () => {
+      const contractFirstERC20BalanceBefore = await firstMockERC20.balanceOf(withdrawTokens.address)
+      const ownerFirstERC20BalanceBefore = await firstMockERC20.balanceOf(owner.address)
+
+      await withdrawTokens
+        .connect(owner)
+        .batchWithdrawERC20([firstMockERC20.address], [contractFirstERC20BalanceBefore.sub(1)])
+
+      expect(await firstMockERC20.balanceOf(withdrawTokens.address)).to.eq(1)
+      expect(await firstMockERC20.balanceOf(owner.address)).to.eq(
+        ownerFirstERC20BalanceBefore.add(contractFirstERC20BalanceBefore.sub(1))
+      )
+    })
+
+    it('withdraws if multiple ERC20 tokens', async () => {
```

Comment:
```suggestion
    it('withdraws different amounts of multiple ERC20 tokens', async () => {
```
You should also use `.sub(2)` for the second amount to ensure that different amounts are recognized.

Reply:
I have re-written the test to ensure that different amounts are withdrawn, using `const ERC20AmountArray = [parseEther('1'), parseEther('2')]`